commands/gist.go

```diff
@@ -0,0 +1,201 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/github/hub/github"
+	"github.com/github/hub/utils"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+var cmdGist = &Command{
+	Run: gist,
+	Usage: `
+gist <GistID> [--no-headers] [<filename>]
```

Comment:
I think printing the contents of an existing gist might not be the functionality that I'd be interested in having in hub right now. I simply don't think it would get much use when compared to creating gists. Furthermore, printing gists with multiple files is awkward, as evident by the need for header lines (which look like git conflicts) and the extra `--no-headers` and `<filename>` arguments.

Would you be open to cutting down the functionality of this command to only support creating gists?

Reply:
The use-cases I'm going here pretty directly need that. Let me lay them out a bit:

* user says "thing isn't working" and does `./my_command | hub gist` and hand that URL to an engineer who can then do `hub gist <thing>`. This is incredibly useful.
* engineer says "hey check out this thing I'm trying to do" and does `cat foo.sh | hub gist` and engineer2 does `hub gist <thing> > /tmp/foo.sh` and runs `foo.sh` to help debug.

The ability to get these back out on the terminal is really really important to the point of writing this module.

I'm not sure why options seem like a downside... configurability seems better to me.